packages/core-data/src/resolvers.js

```diff
@@ -59,9 +63,19 @@ export const getEntityRecord =
 	( kind, name, key = '', query ) =>
 	async ( { select, dispatch } ) => {
 		const configs = await dispatch( getOrLoadEntitiesConfig( kind ) );
```

Comment:
After looking at the code of the whole PR, it seems most of the code to support revisions is already separate from the regular entities code, aside this resolver maybe. So it makes me wonder whether it's actually better to just have a dedicated resolver. Sure the resolver is going to be similar but it will also not contain all these "if"s.